online-store/src/components/startPage.ts

```diff
@@ -117,222 +116,351 @@ const startPageLayout: string =
-        let checkboxCategory = <HTMLElement>document.createElement('div');
-        checkboxCategory.className = 'checkbox-line wrap item-active';
-        checkboxCategory.innerHTML = 
-        `<input type="checkbox" id="${arrayCategory[i]}">
+  for (let i = 0; i < arrayCategory.length; i++) {
+    let checkboxCategory = <HTMLElement>document.createElement("div");
+    checkboxCategory.className = "checkbox-line wrap item-active";
+    checkboxCategory.innerHTML = `<input type="checkbox" id="${arrayCategory[i]}">
         <label for="${arrayCategory[i]}">${arrayCategory[i]}</label>
-        <span>(5/5)</span>`
-        filterCategory.append(checkboxCategory);
-
-    }
-    for (let i=0; i<arrayBrand.length; i++){
-        let checkboxBrand = <HTMLElement>document.createElement('div');
-        checkboxBrand.className = 'checkbox-line wrap item-active';
-        checkboxBrand.innerHTML = 
-        `<input type="checkbox" id="${arrayBrand[i]}">
+        <span>(5/5)</span>`;
+    filterCategory.append(checkboxCategory);
+  }
+  for (let i = 0; i < arrayBrand.length; i++) {
+    let checkboxBrand = <HTMLElement>document.createElement("div");
+    checkboxBrand.className = "checkbox-line wrap item-active";
+    checkboxBrand.innerHTML = `<input type="checkbox" id="${arrayBrand[i]}">
         <label for="${arrayBrand[i]}">${arrayBrand[i]}</label>
-        <span>(5/5)</span>`
-        filterBrand.append(checkboxBrand);
-    }
+        <span>(5/5)</span>`;
+    filterBrand.append(checkboxBrand);
+  }
 
-    let allCards: HTMLCollection = productsCards.children;
+  let allCards: HTMLCollection = productsCards.children;
 
-    if (allCards != undefined) {
+  if (allCards != undefined) {
     for (let i = 0; i < dataProducts.length; i++) {
-        allCards[i].addEventListener("click", (e) => {
-            let eventElem = <HTMLImageElement>e.target;
-            if (eventElem.classList.contains('card-basket-img')){
-                addToCart(eventElem, eventElem.parentElement?.parentElement?.id);
-                console.log(eventElem.parentElement?.parentElement?.id);
-                console.log(idArrayElemAddCart);
-                localStorage.setItem('idArrayCart', idArrayElemAddCart);
-                localStorage.setItem('count', String(count));
-                localStorage.setItem('totalCard', String(summa));
-            } else {
-                productPage(i);
-            }
-        });
+      allCards[i].addEventListener("click", (e) => {
+        let eventElem = <HTMLImageElement>e.target;
+        if (eventElem.classList.contains("card-basket-img")) {
+          addToCart(eventElem, eventElem.parentElement?.parentElement?.id);
+          console.log(eventElem.parentElement?.parentElement?.id);
+          console.log(idArrayElemAddCart);
+          localStorage.setItem("idArrayCart", idArrayElemAddCart);
+          localStorage.setItem("count", String(count));
+          localStorage.setItem("totalCard", String(summa));
```

Comment:
to avoid multiple event listeners, find a parent container (.products-cards) and add **only one listener** on it. Then add a condition to check if e.target element contains class and other code 201-206